modMcf/src/org/aion/mcf/trie/JournalPruneDataSource.java

```diff
@@ -222,29 +291,54 @@ private synchronized void rollback(BH header) {
     }
 
     public int getDeletedKeysCount() {
-        return currentUpdates.deletedKeys.size();
+        lock.readLock().lock();
+        try {
+            return currentUpdates.deletedKeys.size();
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int getInsertedKeysCount() {
-        return currentUpdates.insertedKeys.size();
+        lock.readLock().lock();
+        try {
+            return currentUpdates.insertedKeys.size();
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public Optional<byte[]> get(byte[] key) {
-        return src.get(key);
+        lock.readLock().lock();
+        try {
+            return src.get(key);
+        } catch (Exception e) {
+            throw e;
+        } finally {
```

Comment:
for the following throws(including this) , add an error log before throw if no outside classed catch these exceptions.